### PR TITLE
Chad/#34 event filter screen

### PR DESCRIPTION
**Pull request by chadlagore**
⏳ Status: _Open for visibility_

# 👷 Changes

+ Added a date filter.
+ Added a datetime picker.
+ Updated event redux store on date pick.

# 🏁 Testing Instructions

+ Go to event filter page, click Set Date. Ensure the events in the lister update to your spec.

# ☑️ TODO

- [x] ~Add location filter (the other filtration the backend can handle right now).~

What other filtration do we want? We're only collecting Vancouver on BE right now, we can open that up to BC-wide pretty easily I suppose 🤔. Location isn't very exciting because we're using location services anyways.

### CODE REVIEW

File: src/views/EventFilter.js [hunk 1]
```diff
@@ -1,24 +1,81 @@
...
+        console.log('A date has been picked: ', Moment(date));
+        this._hideDateTimePicker();
+    };
+
+    render() {
         return (
-            <View style={styles.container}>
-                <Text>Filter events by location, price, etc.</Text>
+            <View style={{ flex: 1 }}>
+                <TouchableOpacity onPress={this._showDateTimePicker}>
```
Comment (jordanschalm) [inline src/views/EventFilter.js]: This is probably better as a button, then you don't need the `Text` node

File: src/views/EventFilter.js [hunk 2]
```diff
@@ -1,24 +1,81 @@
...
+        );
+        console.log('A date has been picked: ', Moment(date));
+        this._hideDateTimePicker();
+    };
+
+    render() {
         return (
-            <View style={styles.container}>
-                <Text>Filter events by location, price, etc.</Text>
+            <View style={{ flex: 1 }}>
```
Comment (jordanschalm) [inline src/views/EventFilter.js]: use StyleSheet `styles.container`

File: src/views/EventFilter.js [hunk 3]
```diff
@@ -1,24 +1,81 @@
...
+
+    _handleDatePicked = (date) => {
+        getEvents(
+            // Set start/end window because backend requires both.
+            { ...this.props.location,
+                start_time: Moment(date).format('YYYY-MM-DD'),
+                end_time: Moment(date).add(1, 'days').format('YYYY-MM-DD') }
+            ).then(
+                res => this.props.addEvents(res),
+                err => console.log(err) || props.addEvents([mockEvent])
```
Comment (jordanschalm) [inline src/views/EventFilter.js]: Pretty sure `props` in `props.addEvents` will be undefined here. [ESLint plugin for VSCode will help](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)

Comment (chadlagore) [reply src/views/EventFilter.js]: It updates the events though 🤔

Comment (chadlagore) [reply src/views/EventFilter.js]: I'll get my linter going.

Comment (jordanschalm) [reply src/views/EventFilter.js]: I mean the second one on line 33
`this.props` will work, don't think just `props` will

**Pull request merged**